src/AutoMapper.Extensions.Microsoft.DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -103,14 +103,12 @@ void ConfigAction(IMapperConfigurationExpression cfg)
                 services.AddTransient(type.AsType());
             }
 
-            services.AddSingleton(config);
+            services.AddSingleton<IConfigurationProvider>(sp => new MapperConfiguration(cfg => ConfigAction(sp, cfg)));

```

Comment:
What's the implication here of moving from an instance to a lambda?

Reply:
We are moving the creation of `IConfigurationProvider` from when we call `AddAutoMapper`-extension into when it is used for the first time.